### ISSUE TITLE
ONNX fails to initialize cuDNN during model initialization

### ISSUE THREAD

**Issue by OvervCW** [microsoft/onnxruntime#15355]
### Describe the issue

I'm using the onnxruntime from Triton and while trying to load a model, it failed with the following error:

>2023-04-04 11:54:06.191775568 [E:onnxruntime:, inference_session.cc:1499 operator()] Exception during initialization: /workspace/onnxruntime/onnxruntime/core/providers/cuda/cuda_call.cc:124 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*) [with ERRTYPE = cudnnStatus_t; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] /workspace/onnxruntime/onnxruntime/core/providers/cuda/cuda_call.cc:117 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*) [with ERRTYPE = cudnnStatus_t; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] **CUDNN failure 4: CUDNN_STATUS_INTERNAL_ERROR ; GPU=0 ; hostname=xxx ; expr=cudnnCreate(&cudnn_handle_);**

The process that logged this error had already been running for a few hours, had successfully loaded other models and was performing inference without any issues at the same time that this error appeared. I initially suspected this to be a VRAM or RAM issue, but at the time of this error only 2.3 GiB out of a total of 16 GiB VRAM was in use. The specific model that triggered this error loaded without any issues after restarting the process and only consumed a couple of 100 MiBs of VRAM.

According to the [cuDNN documentation](https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnCreate) this error means that "CUDA resource allocation failed", but I'm not sure what that means, especially when there appears to be plenty of VRAM available.

I'm unsure if this is relevant, but I'm loading models with the following parameters to conserve memory as much as possible:

* `arena_extend_strategy=1` (`kSameAsRequested`)
* `memory.enable_memory_arena_shrinkage=gpu:0`

I'm measuring available VRAM using [nvmlDeviceGetMemoryInfo()](https://docs.nvidia.com/deploy/nvml-api/structnvmlMemory__v2__t.html#structnvmlMemory__v2__t_138b9e3b00a9a3a0eb1ba39bb00502504), which returns `Allocated device memory (in bytes). Note that the driver/GPU always sets aside a small amount of memory for bookkeeping.`.

What could have caused this cuDNN failure?

### To reproduce

Unfortunately I don't know how to reproduce this since I currently don't understand how this issue was caused in the first place.

### Urgency

\-

### Platform

Linux

### OS Version

Ubuntu 20.04.5 LTS (Focal Fossa)

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.13.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

CUDA

### Execution Provider Library Version

CUDA 11.8.0, cuDNN 8.6.0.163

**Comment by tianleiwu** [comment]
If your script imports other packages (like torch, which might use different version of cuDNN and cause confliction), you can try a simple script that only import onnxruntime to see whether session could be created.

**Comment by fanyy94** [comment]
Hi, I am having the same problem. Did you find out what can cause this issue?

**Comment by OvervCW** [comment]
Sorry for the lack of updates. We found out that this is caused by [a problem with Docker + NVIDIA](https://github.com/Azure/AKS/issues/3601) and is not a problem with ONNX.

We confirmed this by the fact that commands like `nvidia-smi` also inside these ONNX containers also started failing the moment this error presented itself, and that this also happens in non-ONNX GPU containers that run TensorFlow Serving, for example.
